core/src/main/java/com/datastax/oss/driver/internal/core/util/concurrent/Debouncer.java

```diff
@@ -107,8 +107,10 @@ public void flushNow() {
     assert adminExecutor.inEventLoop();
     LOG.debug("Flushing now");
     cancelNextFlush();
-    onFlush.accept(coalescer.apply(currentBatch));
-    currentBatch = new ArrayList<>();
+    if (!currentBatch.isEmpty()) {
```

Comment:
In which scenario did you observe that? I'm surprised this can happen, since a flush is only scheduled when the first element is received.

Reply:
I noticed that when I set `metadata.schema.debouncer.window = 0 seconds` in one of my tests that an assertion was failing in `MetadataManager$SingleThreaded.coalesceSchemaRequests`.  I think this is because the code in `acceptSchemaRequest` isn't aware that the debouncer window is 0 and in that case the flush happens right away.

```      
      schemaRefreshDebouncer.receive(future);
      if (flushNow) {
        schemaRefreshDebouncer.flushNow();
      }```

```
java.lang.AssertionError
	at com.datastax.oss.driver.internal.core.metadata.MetadataManager$SingleThreaded.coalesceSchemaRequests(MetadataManager.java:304)
	at com.datastax.oss.driver.internal.core.util.concurrent.Debouncer.flushNow(Debouncer.java:110)
	at com.datastax.oss.driver.internal.core.metadata.MetadataManager$SingleThreaded.acceptSchemaRequest(MetadataManager.java:296)
	at com.datastax.oss.driver.internal.core.metadata.MetadataManager$SingleThreaded.access$1100(MetadataManager.java:224)
	at com.datastax.oss.driver.internal.core.metadata.MetadataManager.lambda$refreshSchema$5(MetadataManager.java:176)
	at com.datastax.oss.driver.internal.core.util.concurrent.RunOrSchedule.on(RunOrSchedule.java:55)
	at com.datastax.oss.driver.internal.core.metadata.MetadataManager.refreshSchema(MetadataManager.java:176)
	at com.datastax.oss.driver.internal.core.DefaultCluster$SingleThreaded.afterInitialNodeListRefresh(DefaultCluster.java:219)
```

Reply:
I see, the problem comes from this method:
```java
    private void acceptSchemaRequest(CompletableFuture<Metadata> future, boolean flushNow) {
      assert adminExecutor.inEventLoop();
      schemaRefreshDebouncer.receive(future);
      if (flushNow) {
        schemaRefreshDebouncer.flushNow();
      }
    }
```
Since the window is 0, `receive` has already flushed, so when we do the `flushNow` there is nothing left. I think it would be better to test the batch size in `flushNow`.

Reply:
> I think it would be better to test the batch size in flushNow.

Oops, your fix *is* in flushNow, ignore this :man_facepalming: